Fri 7th July 2006 Stuart Hughes <stuarth@freescale.com>

These are changes I've made to the client and server scripts/configs to
use more generic defaults that work out of the box.


diff --exclude CVS -uNr dhcp-3.0.3b1/client/scripts/linux dhcp-3.0.3b1.modified/client/scripts/linux
--- dhcp-3.0.3b1/client/scripts/linux	2002-11-15 01:09:09.000000000 +0000
+++ dhcp-3.0.3b1.modified/client/scripts/linux	2006-07-07 11:55:57.000000000 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # dhclient-script for Linux. Dan Halbert, March, 1997.
 # Updated for Linux 2.[12] by Brian J. Murrell, January 1999.
 # No guarantees about this. I'm a novice at the details of Linux
@@ -23,11 +23,20 @@
 # of the $1 in its args.
 
 make_resolv_conf() {
-  if [ "x$new_domain_name" != x ] && [ x"$new_domain_name_servers" != x ]; then
-    echo search $new_domain_name >/etc/resolv.conf
+  if [ x$new_domain_name != x -o x$new_domain_name_servers != x ]
+  then
+    echo "# autogenerated by /sbin/dhclient-script" > /etc/resolv.conf
     chmod 644 /etc/resolv.conf
-    for nameserver in $new_domain_name_servers; do
-      echo nameserver $nameserver >>/etc/resolv.conf
+  fi
+  if [ x$new_domain_name != x ]
+  then
+    echo "search $new_domain_name" >> /etc/resolv.conf
+  fi
+  if [ x$new_domain_name_servers != x ]
+  then
+    for nameserver in $new_domain_name_servers
+    do
+      echo "nameserver $nameserver" >> /etc/resolv.conf
     done
   fi
 }
diff --exclude CVS -uNr dhcp-3.0.3b1/tests/config_script/dhclient.conf dhcp-3.0.3b1.modified/tests/config_script/dhclient.conf
--- dhcp-3.0.3b1/tests/config_script/dhclient.conf	2006-07-07 15:27:19.000000000 +0100
+++ dhcp-3.0.3b1.modified/tests/config_script/dhclient.conf	2006-07-07 11:26:42.000000000 +0100
@@ -1,32 +1,32 @@
-timeout 1;
-retry 60;
-reboot 10;
-select-timeout 5;
-initial-interval 2;
-reject 192.33.137.209;
+#timeout 1;
+#retry 60;
+#reboot 10;
+#select-timeout 5;
+#initial-interval 2;
+#reject 192.33.137.209;
 option SIP code 120=string;
 interface "eth0" {
            send host-name "dhcphost";
-           send dhcp-client-identifier 00:04:9f:96:9c:4d;
-	   send dhcp-lease-time 3600;
-           supersede domain-name "emsys.net";
-           prepend domain-name-servers 203.145.184.13;
+#           send dhcp-client-identifier 00:04:9f:96:9c:4d;
+#	   send dhcp-lease-time 3600;
+#           supersede domain-name "emsys.net";
+#           prepend domain-name-servers 203.145.184.13;
            request subnet-mask, broadcast-address, time-offset, routers,
                 domain-name,domain-name-servers, host-name,SIP;
            require subnet-mask, domain-name-servers;
       }
 
 
-lease {
-  interface "eth0";
-  fixed-address 192.168.1.168;
-  medium "eth0 link0 link1";
-  option host-name "dhcphost";
-  option subnet-mask 255.255.255.0;
-  option broadcast-address 192.168.1.255;
-  option routers 192.168.1.1;
-  option domain-name-servers 203.145.184.13;
-  renew 2 2005/1/12 00:00:01;
-  rebind 2 2005/1/12 00:00:01;
-  expire 2 2005/7/12 00:00:01;
-}
+#lease {
+#  interface "eth0";
+#  fixed-address 192.168.1.168;
+#  medium "eth0 link0 link1";
+#  option host-name "dhcphost";
+#  option subnet-mask 255.255.255.0;
+#  option broadcast-address 192.168.1.255;
+#  option routers 192.168.1.1;
+#  option domain-name-servers 203.145.184.13;
+#  renew 2 2005/1/12 00:00:01;
+#  rebind 2 2005/1/12 00:00:01;
+#  expire 2 2005/7/12 00:00:01;
+#}
